### PR TITLE
feat: remove Token/IResolvable API

### PR DESCRIPTION
**Pull request by RomainMuller**
BREAKING CHANGE: The `Token` class as well as all related elements
(`IResolvable`, etc...) were removed from this release. The feature may
be re-introduced in the future, but was pulled due to causing conflicts
with overlapping implementations in some consumers.

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: lib/construct.ts [hunk 1]
```diff
@@ -76,10 +75,6 @@ export class Node {
       // This is a root construct.
       this.id = id;
     }
-
-    if (Token.isUnresolved(id)) {
```
Comment (eladb) [inline lib/construct.ts:80]: I wonder if we should allow users to pass in an `identityValidator` (similar to `nodeFactory`) so we can implement this in the AWS CDK to protect against using tokens in construct IDs?

Comment (RomainMuller) [reply lib/construct.ts:80]: Sounds interesting... let me look into it.

**Comment by RomainMuller** [comment]
Not sure how I should be rolling over the breaking changes blowing the PR validation build. @eladb  - should I also bump the version there?

**Comment by eladb** [comment]
> Not sure how I should be rolling over the breaking changes blowing the PR validation build. @eladb  - should I also bump the version there?

I am okay with bumping to 3.0.0 in the same PR

**Comment by RomainMuller** [comment]
So I did, but the test still fails... Apparently it doesn't seem concerned with the fact I performed a major bump 🤷🏻‍♂️

File: lib/construct.ts [hunk 2]
```diff
@@ -715,3 +706,33 @@ export interface INodeFactory {
...
+   * @param id the construct id (__after__ sanitization).
+   * @throws if `id` is not a valid construct ID.
+   */
+  validateConstructId(id: string): void;
+}
+
+/**
+ * The default implementation of `INodeFactory`.
+ */
+export class DefaultNodeFactory implements INodeFactory {
```
Comment (eladb) [inline lib/construct.ts:725]: Don't export this please.

File: lib/construct.ts [hunk 3]
```diff
@@ -54,15 +53,19 @@ export class Node {
   private readonly _context: { [key: string]: any } = { };
   private readonly _metadata = new Array<MetadataEntry>();
   private readonly _dependencies = new Set<IConstruct>();
+  private readonly _idValidator?: IConstructIdValidator;
   private readonly invokedAspects: IAspect[] = [];
   private _defaultChild: IConstruct | undefined;
 
-  constructor(private readonly host: Construct, scope: IConstruct, id: string) {
+  constructor(private readonly host: Construct, scope: IConstruct, id: string, idValidator?: IConstructIdValidator) {
```
Comment (eladb) [inline lib/construct.ts:60]: I rather leave the validation in `Construct` and not pass this to `Node`.

File: package.json [hunk 4]
```diff
@@ -39,8 +39,8 @@
     ],
     "coverageThreshold": {
       "global": {
-        "lines": 78,
-        "statements": 78
+        "lines": 70,
```
Comment (eladb) [inline package.json:42]: Why?

Comment (RomainMuller) [reply package.json:42]: Dropped some highly-covered code... So the remaining code is less covered :(

Comment (RomainMuller) [reply package.json:42]: Dropping `lib/stack-traces.ts` (unexpected, only used by Tokens) allows dialing back up to `77%`.

File: lib/construct.ts [hunk 5]
```diff
@@ -533,13 +521,21 @@ export class Construct implements IConstruct {
    * @param options Options
    */
   constructor(scope: Construct, id: string, options: ConstructOptions = { }) {
+    // validate the construct ID
+    const validator = options.constructIdValidator ?? (scope as any)?.[CONSTRUCT_ID_VALIDATOR_SYMBOL];
```
[the remark this answers is not in the document]
Comment (RomainMuller) [reply lib/construct.ts:525]: That felt convenient to me - not having to specify a validator beyond the root of the construct tree?

[the remark this answers is not in the document]
Comment (eladb) [reply lib/construct.ts:525]: Yeah, let's get rid of this id validator shit.

**Comment by eladb** [comment]
@RomainMuller I disabled API compat check temporarily to allow the MV bump

**Pull request merged**